packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -352,18 +369,20 @@ export class ElementController<TElement extends HTMLElement = HTMLElement>
             this.view.bind(this.source);
         }
 
-        this.setIsConnected(true);
+        this.stage = Stages.connected;
+        Observable.notify(this, isConnectedPropertyName);
     }
 
     /**
      * Runs disconnected lifecycle behavior on the associated element.
      */
     public disconnect(): void {
-        if (!this._isConnected) {
+        if (!this.isConnected) {
```

Comment:
Should this also be expanded out like the `connect` method to check `disconnected` and `disconnecting`?

Reply:
That makes sense to me. If it's an "ing" stage then that could be a double entry, so we want to avoid that as well.